### PR TITLE
i18n(pt-BR) Add `migrate.md` translation

### PR DESCRIPTION
**Pull request by Yan-Thomas**
#### What kind of changes does this PR include?

- [ ] Minor content fixes (broken links, typos, etc.)
- [ ] New or updated content
- [x] Translated content
- [ ] Changes to the docs site code
- [ ] Something else!

#### Description

PT-BR translation for`migrate.md`

### CODE REVIEW

File: src/pages/pt-BR/migrate.md [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+- Move `Astro.request.canonicalURL` para `Astro.canonicalURL`.
+
+Leia [RFC0018](https://github.com/withastro/rfcs/blob/main/proposals/0018-astro-request.md) para mais contexto sobre estas mudanças.
+
+
+### Outras Mudanças
+
+- Melhora a API de `Astro.slots` para suportar a passagem de argumentos para slots baseados em funções. Isto permite componentes utilitários mais ergonômicos que aceitam uma função de callback como um filho.
+- Atualiza a formatação da interface de linha de comando, especialmente de relatório de erros.
+- Atualiza `@astrojs/compiler`, arrumando alguns bugs relacionadas a utilização de RegExp no frontmatter.
```
Comment (victor0x16) [inline src/pages/pt-BR/migrate.md:101]:
```suggestion
- Atualiza `@astrojs/compiler`, corrigindo alguns bugs relacionadas a utilização de RegExp no frontmatter.
```

File: src/pages/pt-BR/migrate.md [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+
+Anteriormente, React, Preact, Svelte e Vue eram todos inclusos no Astro por padrão. A partir da v0.25.0, Astro não virá mais com qualquer renderizador integrado. Se você não tinha uma entrada `renderers` já definida no seu projeto, você agora precisará instalar esses frameworks por si mesmo.
+
+Leia nosso [guia passo-a-passo](/pt-BR/guides/integrations-guide/) para aprender como adicionar uma nova integração Astro para o(s) framework(s) que você atualmente utiliza.
+
+#### Descontinuado: Renderers
+
+> *Leia esta seção se você já tiver "renderers" customizados definidos no seu arquivo de configuração.*
+
+O novo sistema de integraçõea substituído o antigo sistema de `renderers`, incluindo os pacotes `@astrojs/renderer-*` publicados no npm. De agora em diante, `@astrojs/renderer-react` se torna `@astrojs/react`, `@astrojs/renderer-vue` se torna `@astrojs/vue`, e por aí vai.
```
Comment (victor0x16) [inline src/pages/pt-BR/migrate.md:121]:
```suggestion
O novo sistema de integrações substitui o antigo sistema de `renderers`, incluindo os pacotes `@astrojs/renderer-*` publicados no npm. De agora em diante, `@astrojs/renderer-react` se torna `@astrojs/react`, `@astrojs/renderer-vue` se torna `@astrojs/vue`, e por aí vai.
```

File: src/pages/pt-BR/migrate.md [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+
+
+**Para migrar:** atualize Astro para a `v0.25.0` e então execute `astro dev` ou `astro build` com o seu antigo arquivo de configuração contendo a desatualizada configuração `"renderers"`. Você irá imediatamente ver um aviso te dizendo as exatas modificações que você precisa fazer em seu arquivo `astro.config.mjs`, baseado na configuração atual. VOcê também pode atualizar seus pacotes por si mesmo, utilizando a tabela abaixo.
```
Comment (victor0x16) [inline src/pages/pt-BR/migrate.md:124]:
```suggestion
**Para migrar:** atualize Astro para a `v0.25.0` e então execute `astro dev` ou `astro build` com o seu antigo arquivo de configuração contendo a desatualizada configuração `"renderers"`. Você irá imediatamente ver um aviso te dizendo as exatas modificações que você precisa fazer em seu arquivo `astro.config.mjs`, baseado na configuração atual. Você também pode atualizar seus pacotes por si mesmo, utilizando a tabela abaixo.
```

File: src/pages/pt-BR/migrate.md [hunk 4]
```diff
@@ -0,0 +1,510 @@
...
+
+Para um passo-a-passo mais detalhado, veja nosso [guia passo-a-passo](/pt-BR/guides/integrations-guide/) para aprender como substituir renderizadores existentes com uma nova integração de framework no Astro.
```
Comment (victor0x16) [inline src/pages/pt-BR/migrate.md:126]:
```suggestion
Para um passo-a-passo mais detalhado, veja nosso [guia passo-a-passo](/pt-BR/guides/integrations-guide/) para aprender como substituir renderers existentes com uma nova integração de framework no Astro.
```

File: src/pages/pt-BR/migrate.md [hunk 5]
```diff
@@ -0,0 +1,510 @@
...
++ import react from '@astrojs/react';
+
+export default {
+-   renderers: ['@astrojs/renderer-lit', '@astrojs/renderer-react'],
++   integrations: [lit(), react()],
+}
+```
+
+
+| Renderizadores descontinuados no npm | Integrações no npm para a v0.25+ |
```
Comment (victor0x16) [inline src/pages/pt-BR/migrate.md:147]:
```suggestion
| Renderers descontinuados no npm | Integrações no npm para a v0.25+ |
```

File: src/pages/pt-BR/migrate.md [hunk 6]
```diff
@@ -0,0 +1,510 @@
...
+| @astrojs/renderer-preact    | @astrojs/preact            |
+| @astrojs/renderer-solid     | @astrojs/solid-js          |
+| @astrojs/renderer-vue       | @astrojs/vue               |
+| @astrojs/renderer-svelte    | @astrojs/svelte            |
+
+#### Lidando com Dependências de Pares
+
+> *Leia esta seção se: Você está no Node v14 **ou** se você utiliza algum gerenciador de pacotes que não seja o npm.*
+
+Diferente dos antigos renderizadores, integrações não mais marcam os próprios frameworks ("react", "svelte", "vue", etc.) como dependências diretas da integração. NO lugar, você deve agora instalar os pacotes dos seus frameworks *assim como* suas integrações.
```
Comment (victor0x16) [inline src/pages/pt-BR/migrate.md:159]:
```suggestion
Diferente dos antigos renderers, integrações não mais marcam os próprios frameworks ("react", "svelte", "vue", etc.) como dependências diretas da integração. No lugar, você deve agora instalar os pacotes dos seus frameworks *assim como* suas integrações.
```

File: src/pages/pt-BR/migrate.md [hunk 7]
```diff
@@ -0,0 +1,510 @@
...
+
+```diff
+# Exemplo: Instale integrações e frameworks juntos
+- npm install @astrojs/react
++ npm install @astrojs/react react react-dom
+```
+
+Se você ver um aviso `"Cannot find package 'react'"` (ou similar) quando você inicia o Astro, isso significa que você precisa instalar tal pacote em seu projeto. Veja nossa [nota em dependências de pares](/pt-BR/guides/integrations-guide/#lidando-com-dependências-de-integrações) no guia de integrações para mais informações.
+
+Se você estiver utilizando `npm` e Node v16+, então isso pode ser lidado automaticamente para você pelo `npm`, desde a última versão do `npm` (v7+) ele instala dependências de pares assim automaticamente para você. Nesse caso, instalar um framework como "react" em seu projeto é opcional mas ainda uma etapa recomendada.
```
Comment (victor0x16) [inline src/pages/pt-BR/migrate.md:169]:
```suggestion
Se você estiver utilizando `npm` e Node v16+, então isso pode ser feito automaticamente para você pelo `npm`, desde a última versão do `npm` (v7+) ele instala dependências de pares assim automaticamente para você. Nesse caso, instalar um framework como "react" em seu projeto é opcional mas ainda uma etapa recomendada.
```

File: src/pages/pt-BR/migrate.md [hunk 8]
```diff
@@ -0,0 +1,510 @@
...
+**Quando utilizar isto:** Se o seu arquivo JavaScript está dentro de `public/`.
+
+Você pode refenciar qualquer arquivo dentro do diretório `public/` com uma URL de caminho absoluto nos templates de seus componentes Astro. Esta é uma boa opção por padrão para scripts externos, já que isto te permite controlar a tag `<script>` na página por si mesmo.
+
+Note que esta abordagem pula o processamento, o bundle e otimizações de JavaScript que são providenciadas pelo Astro quando você utiliza o método com `import` descrito acima. Porém, isso pode ser preferível para quaisquer scripts externos que já foram publicados e minificados separadamente do Astro. Se o seu script foi baixado de uma fonte externa, então este método é provavelmente preferível.
+
+**2. Importação ESM via `<script hoist>`**
+
+**Exemplo:** `<script hoist>import './algum-script-externo.js';</script>`
+**Quando utilizar isto:** Se o seu script externo está dentro de `src/` e_ ele suporta o tipo de módulo ESM.
```
Comment (victor0x16) [inline src/pages/pt-BR/migrate.md:256]:
```suggestion
**Quando utilizar isto:** Se o seu script externo está dentro de `src/` e ele suporta o tipo de módulo ESM.
```

File: src/pages/pt-BR/migrate.md [hunk 9]
```diff
@@ -0,0 +1,510 @@
...
+
+```
+Preprocessor dependency "sass" not found. Did you install it?
+```
+
+Em nossa missão para reduzir o tamanho de npm install, nós movemos o [Sass](https://sass-lang.com/) como uma dependência opcional. Se você utiliza Sass em seu projeto, você vai querer se certificar de que executou `npm install sass --save-dev` para salvá-lo como uma dependência.
+
+### Descontinuado: HTML não Sanitizado
+
+No Astro v0.23+, conteúdo HTML não sanitizado em expressões é agora não é mais possível.
```
Comment (victor0x16) [inline src/pages/pt-BR/migrate.md:323]:
```suggestion
No Astro v0.23+, conteúdo HTML não sanitizado em expressões não é mais possível.
```

File: src/pages/pt-BR/migrate.md [hunk 10]
```diff
@@ -0,0 +1,510 @@
...
+export default {
+  vite: {
+    plugins: [imagetools()],
+  },
+};
+```
+
+Para aprender mais sobre plugins Vite, por favor visite seu [guia de plugins](https://vitejs.dev/guide/using-plugins.html).
+
+#### Mudanças do Vite para Renderizadores
```
Comment (victor0x16) [inline src/pages/pt-BR/migrate.md:389]:
```suggestion
#### Mudanças do Vite para Renderers
```

**Review by victor0x16 — CHANGES_REQUESTED** [review]
@Yan-Thomas , my two cents :)

**Comment by Yan-Thomas** [comment]
Thanks @victor0x16! :rocket:

**Review by victor0x16 — APPROVED** [review]
LGTM.

**Pull request merged**